python/lsst/ts/standardscripts/scheduler/base_set_desired_state.py

```diff
@@ -0,0 +1,214 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["BaseSetDesiredState"]
+
+import abc
+import types
+import typing
+import asyncio
+
+from lsst.ts import salobj
+
+
+class BaseSetDesiredState(salobj.BaseScript):
+    """A base script that implements setting the desired state for the
+    Scheduler.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+    descr : `str`
+        Script description.
+    scheduler_index : `int`
+        Index of the Scheduler to enable.
+    """
+
+    def __init__(self, index: int, descr: str, scheduler_index: int) -> None:
```

Comment:
I strongly suggest formatting descr in this constructor instead of taking it as an argument for all of these new base classes.

That simplifies use of the class and avoids strings that are wrong (e.g. due to copy/paste errors).

It should be trivial to do as long as you cast the scheduler_index (and desired_state, for this one) into the appropriate enums, so you can use the .name property. That cast has the added advantage of catching invalid values and the somewhat questionable advantage of allowing integer arguments.